File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/MedicationAdministration.ts
import { Annotation } from "./Annotation";
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Quantity } from "./Quantity";
import { DomainResource } from "./DomainResource";
import { Ratio } from "./Ratio";
import { Reference } from "./Reference";
import { Identifier } from "./Identifier";
import { BackboneElement } from "./BackboneElement";

export type MedicationAdministrationDosage = BackboneElement & {
    dose?: Quantity;
    rate?: Ratio | Quantity;
    site?: CodeableConcept;
    text?: string;
    route?: CodeableConcept;
    method?: CodeableConcept;
};

export type MedicationAdministrationPerformer = BackboneElement & {
    actor: Reference;
    function?: CodeableConcept;
};

export type MedicationAdministration = DomainResource & {
    category?: CodeableConcept;
    request?: Reference;
    eventHistory?: Reference[];
    dosage?: MedicationAdministrationDosage;
    instantiates?: string[];
    reasonCode?: CodeableConcept[];
    statusReason?: CodeableConcept[];
    note?: Annotation[];
    supportingInformation?: Reference[];
    status: string;
    effective?: string | Period;
    identifier?: Identifier[];
    context?: Reference;
    device?: Reference[];
    partOf?: Reference[];
    subject: Reference;
    performer?: MedicationAdministrationPerformer[];
    medication?: CodeableConcept | Reference;
    reasonReference?: Reference[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/MedicationAdministration.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Annotation
from base import Period
from base import CodeableConcept
from base import Quantity
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

@dataclass(kw_only=True)
class MedicationAdministrationDosage(BackboneElement):
    dose: Optional[Quantity] = None
    site: Optional[CodeableConcept] = None
    text: Optional[str] = None
    route: Optional[CodeableConcept] = None
    method: Optional[CodeableConcept] = None
    rate_ratio: Optional[Ratio] = None
    rate_quantity: Optional[Quantity] = None

@dataclass(kw_only=True)
class MedicationAdministrationPerformer(BackboneElement):
    actor: Reference
    function: Optional[CodeableConcept] = None

@dataclass(kw_only=True)
class MedicationAdministration(DomainResource):
    category: Optional[CodeableConcept] = None
    request: Optional[Reference] = None
    event_history: Optional[List[Reference]] = field(default_factory=list)
    dosage: Optional[MedicationAdministrationDosage] = None
    instantiates: Optional[List[str]] = field(default_factory=list)
    reason_code: Optional[List[CodeableConcept]] = field(default_factory=list)
    medication_codeable_concept: Optional[CodeableConcept] = None
    status_reason: Optional[List[CodeableConcept]] = field(default_factory=list)
    note: Optional[List[Annotation]] = field(default_factory=list)
    supporting_information: Optional[List[Reference]] = field(default_factory=list)
    effective_date_time: Optional[str] = None
    status: str
    identifier: Optional[List[Identifier]] = field(default_factory=list)
    context: Optional[Reference] = None
    device: Optional[List[Reference]] = field(default_factory=list)
    medication_reference: Optional[Reference] = None
    part_of: Optional[List[Reference]] = field(default_factory=list)
    subject: Reference
    performer: Optional[List[MedicationAdministrationPerformer]] = field(default_factory=list)
    effective_period: Optional[Period] = None
    reason_reference: Optional[List[Reference]] = field(default_factory=list)